pynxtools_xps/scienta/scienta_reader.py

```diff
@@ -528,67 +403,207 @@ def _parse_region(self, region_id):
         region.validate_types()
 
         region_dict = {**self.header.dict(), **region.dict()}
-
-        for key in region_dict.copy().keys():
-            if "_units" in key:
-                new_key = key.replace("_units", "/@units")
-                region_dict[new_key] = region_dict.pop(key)
+        region_dict["intensity/@units"] = "counts"
 
         self.spectra.append(region_dict)
 
-    def _check_valid_value(self, value: Union[str, int, float, bool, np.ndarray]):
+
+class ScientaIgorParser:
+    """Parser for Scienta TXT exports."""
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self):
+        self.lines: List[str] = []
+        self.spectra: List[Dict[str, Any]] = []
+
+    def parse_file(self, file: Union[str, Path]):
         """
-        Check if a string or an array is empty.
+        Reads the igor binarywave files and returns a list of
+        dictionary containing the wave data.
 
         Parameters
         ----------
-        value : obj
-            For testing, this can be a str or a np.ndarray.
+        file : str
+            Filepath of the TXT file to be read.
 
         Returns
         -------
-        bool
-            True if the string or np.ndarray is not empty.
+        self.spectra
+            Flat list of dictionaries containing one spectrum each.
 
         """
-        for datatype in [str, int, float]:
-            if isinstance(value, datatype) and value:
-                return True
-        if isinstance(value, bool):
-            return True
-        if isinstance(value, np.ndarray) and value.size != 0:
-            return True
-        return False
-
-    def _get_key_value_pair(self, line: str):
+        ibw = binarywave.load(file)
+        ibw_version, wave = ibw["version"], ibw["wave"]
+
+        notes = self._parse_note(wave["note"])
+
+        data_unit_label, data_unit = self._parse_unit(wave["data_units"])
+        dimension_unit_label, dimension_unit = self._parse_unit(wave["dimension_units"])
+
+        wave_header = wave["wave_header"]
+        data = wave["wData"]
+
+        # Not needed at the moment
```

Comment:
Maybe it's good to state here why it is kept. Something like `TODO: Add support for formulas` or so